### PR TITLE
Add releases.md

### LINKED ISSUE #2700: Document release process
Ref: https://github.com/stylelint/stylelint/pull/2661#issuecomment-311780010

### PR DESCRIPTION
**Pull request by jeddy3**
Closes #2700

@hudochenkov, @ntwb @evilebottnawi If you’d like access to the npm package for performing releases, please pop your npm usernames below and I’ll added you. Likewise, if you’d like access to the stylelint twitter account you’ll need to drop an email address or something so I can send the access details over.

### CODE REVIEW

**Comment by ntwb** [comment]
We should make note that you need a GitHub API token

Because of issues I had for a while I was not always reliably exporting my GitHub API token to my shell, so when I ran `npm release` for `stylelint-config-wordpress` it would fail half way through, as such I [changed the npm](https://github.com/WordPress-Coding-Standards/stylelint-config-wordpress/blob/master/package.json#L58) script to `"release": "npmpub --verbose",` for added debugging, this allowed me to then export my GitHub API token and to manually step through the changes that failed in the terminal 😖 

The `npmpub` module also [comes with a](https://www.npmjs.com/package/npmpub#options)  `--dry` option to perform a _dry test run_ that is useful, per the above I occasionally hack that into my `package.json` to confirm everything is right to go 😬 

Do you think it's worth updating the stylelint npm tasks to include the `--verbose` flag, and a new dry run type task like `"release:dryrun": "npmpub --verbose -dry", or just include some of this in the proposed PR docs here?

**Comment by jeddy3** [comment]
> Do you think it's worth updating the stylelint npm tasks to include the --verbose flag, and a new dry run type task like `"release:dryrun": "npmpub --verbose -dry",

If doing so will help catch missing GitHub API tokens, then I think it's worth it. 

I've added a `dry-release` task (as I believe the colon notation is for tasks that are intended to be run in parallel e.g. `lint:*`) and added an additional step to the process in the `.md` file.

I'll add the task to the `stylelint-config-*` repos too.

**Comment by ntwb** [comment]
p.s. Happy to be added as a _backup user_ for npm and Twitter if you want another user with access: 

NPM: `netweb` Email: `stephen@netweb.com.au`

File: docs/developer-guide/releases.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+1.  Ensuring the stylelint org's `stylelint-config-*` shareable configs are compatible with the release.
+2.  Updating the online demo at [https://stylelint.io/demo](https://stylelint.io/demo) to use the release.
+3.  Updating the documentation at [https://stylelint.io](https://stylelint.io) to use the release.
+4.  Sending out a tweet.
+
+## Process
+
+1.  Locally test `master` in the `stylelint-config-*` shareable configs repos.
+2.  Locally test `master` in the `stylelint.io` repo.
+3.  Locally test `master` in the `stylelint-demo` repo.
```
Comment (hudochenkov) [inline docs/developer-guide/releases.md]: What does it mean? `npm i stylelint/stylelint && npm test` in every repo?

File: docs/developer-guide/releases.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    6.  Run `npm run dry-release`.
+    7.  Run `npm run release`.
+    8.  Go to [https://www.npmjs.com/package/stylelint](https://www.npmjs.com/package/stylelint) and confirm the package was published correctly.
+    9.  Go to [https://github.com/stylelint/stylelint/releases](https://github.com/stylelint/stylelint/releases) and confirm the release was created correctly.
+5.  If a new version of any `stylelint-config-*` is required, repeat step 3 for that repo.
+6.  Update the online demo by changing to the `stylelint-demo` repo:
+    1.  Run `npm install -S stylelint@latest`
+    2.  Run `npm test`
+    3.  Commit and _push up_ these changes.
+    1.  Go to [https://stylelint.io/demo](https://stylelint.io/demo) and confirm the update was automatically deployed.
```
Comment (hudochenkov) [inline docs/developer-guide/releases.md]: How to verify that?

Comment (jeddy3) [inline docs/developer-guide/releases.md]: Check that a feature added or bug fixed in the release works now works at https://stylelint.io/demo

File: docs/developer-guide/releases.md (hunk 1 above)
Comment (jeddy3) [inline docs/developer-guide/releases.md]: Yep

**Comment by jeddy3** [comment]
> When do we need a GitHub API token?

You'll get a warning you do the dry run.

**Pull request merged**